lib/routes/routeBackbeat.js

```diff
@@ -373,20 +380,78 @@ function completeMultipartUpload(request, response, log, callback) {
         }
         return multipleBackendGateway.completeMPU(request.objectKey, uploadId,
             storageLocation, { Part }, undefined,
-            request.bucketName, log, err => {
+            request.bucketName, log, (err, retrievalInfo) => {
                 if (err) {
                     log.error('error completing MPU', {
                         error: err,
                         method: 'completeMultipartUpload',
                     });
                     return callback(err);
                 }
-                return _respond(response, {}, log, callback);
+                const dataRetrievalInfo = {
+                    versionId: retrievalInfo.dataStoreVersionId,
+                };
+                return _respond(response, dataRetrievalInfo, log, callback);
             });
     });
     return undefined;
 }
 
+function putObjectTagging(request, response, log, callback) {
+    const err = _checkMultipleBackendRequest(request, log);
+    if (err) {
+        return callback(err);
+    }
+    const storageLocation = request.headers['x-scal-storage-class'];
+    const dataStoreVersionId = request.headers['x-scal-data-store-version-id'];
+    const data = [];
+    request.on('data', chunk => data.push(chunk));
+    request.on('end', () =>
+        parseTagXml(Buffer.concat(data), log, (err, tags) => {
```

Comment:
> If totalLength is not provided, it is calculated from the Buffer instances in list. This however causes an additional loop to be executed in order to calculate the totalLength, so it is faster to provide the length explicitly if it is already known.

You should aggregate each chunk's length and then provide it to the Buffer.concat method.

Reply:
Updated.